vscode-lean4/src/infoview.ts

```diff
@@ -96,7 +96,17 @@ export class InfoProvider implements Disposable {
         sendClientRequest: async (uri: string, method: string, params: any): Promise<any> => {
             const client = this.clientProvider.findClient(uri);
             if (client) {
-                return client.sendRequest(method, params);
+                try {
+                    const result = await client.sendRequest(method, params);
+                    return result
+                } catch (ex) {
+                    if (ex.code === -32901 || ex.code === -32902) {
+                        // ex codes related with worker exited or crashed
+                        console.log(`Lean worker exited or crashed: ${ex.message}`)
+                        const activeClient = client === this.clientProvider.getActiveClient();
+                        await this.onActiveClientStopped(client, activeClient, 'Lean worker exited or crashed: ', ex.message as string)
```

Comment:
Yes, you are right we piggy backed the server stopped machinery for worker stopped because for a single file edit scenario the user cannot really see the difference, all they see is that the infoview stops working and the only way we have of fixing it is the sledge hammer of "Restart Server", so since the user experience is basically the same we saw reason not to reuse the same code we already had for server stopped.

Reply:
Oh, rats, it would have been nice to know this sooner.  Mariana and I have not been using "Refresh File Dependencies" in all our testing of this scenario and didn't realize that would fix the missing worker problem.   Gabriel are you saying this PR needs to change the error message terminology accordingly and use "Refresh File Dependencies" as the fix instead of "Restart Server" ?   Do you have some recommended language for the error message and for the restart button that would make sense to folks?  Something like:

message: "The Lean language service has stopped processing this file"
reason: ...
button: "Restart the Lean language service on this file" 

?

Reply:
I honestly thought that this was obvious, as that command is the only way to restart a single file and I use it every few minutes. :facepalm:  But going again over the documentation, I can see how you missed it.  The name is misleading, and the readme even calls it a "work around for [technical issue]".  See also: https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Renaming.20.22refresh.20file.20dependencies.22.3F

> Restart the Lean language service on this file

I don't think we ever use the term "language service" in user-facing text.  Usually it's "Lean server", so I'd go with "Restart the Lean server for this file".

Reply:
It makes sense now, not knowing much about how the LSP works, somehow I thought refresh file dependencies was doing more that just restarting the worker process, like there were some sort of caches of dependencies that needed updating and that would fix all open files.  So this means if I have lots of files open and I change a core type they are all using then I have to run refresh file dependencies N times across all N open files?  If so, that kinda sucks...